contrib/firebaseio/src/test/java/utils/LoggingArrayList.java

```diff
@@ -0,0 +1,40 @@
+/**
+ * Copyright (c) 2015 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package utils;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import org.slf4j.Logger;
+
+import java.util.ArrayList;
+
+/**
+ * Logs all the added entries!
+**/
+public class LoggingArrayList<T> extends ArrayList<T> {
```

Comment:
Does this need to be a top-level public class, or can you just nest it as a `private static class` inside the test file that uses it? Seems like a weird thing to add to your public API.

(Could it at least be package-private if not private inner?)